### PR TITLE
Improved Virtualization Guide for SLES

### PR DESCRIPTION
**Pull request by lightlace**
I improved the documentation about the configuration in libvirt, especially the configuration with the Virtual Machine Manager and the documentation about virsh.

### CODE REVIEW

File: xml/libvirt_configuration_virsh.xml [hunk 1]
```diff
@@ -0,0 +1,1231 @@
...
+<screen><prompt>tux &gt; </prompt><command>sudo virsh nodedev-detach pci_0000_04_10_0</command></screen>
+    </step>
+    <step>
+     <para>
+      Add the VF interface to an existing VM Guest:
+     </para>
+<screen><prompt>tux &gt; </prompt><command>sudo virsh attach-device <replaceable>GUEST</replaceable> /tmp/vf-interface.xml --<replaceable>OPTION</replaceable></command></screen>
+     <para>
+      <replaceable>GUEST</replaceable> needs to be replaced by the domain
+      name, id or uuid of the VM Guest and
```
Comment (taroth21) [inline xml/libvirt_configuration_virsh.xml]: 'id or uuid' -> 'ID' or 'UUID' (?).
Split into two sentences: 
'[...] of the VM Guest.  '--<replaceable>OPTION</replaceable> can be one of the following: [...]'

File: xml/libvirt_configuration_virsh.xml [hunk 2]
```diff
@@ -0,0 +1,1231 @@
...
+    The guest then references this network, and each time it is started, a
+    single VF is dynamically allocated to it. When the guest is stopped, the
+    VF is returned to the pool, available for another guest.
+   </para>
+   <sect3 xml:id="libvirt.config.io.pool.host">
+    <title>Defining Network with Pool of VFs on VM Host Server</title>
+    <para>
+     The following example of network definition creates a pool of all VFs
+     for the <xref linkend="vt.io.sriov"/> device with its physical function
+     (PF) at the network interface eth0 on the host:
```
Comment (taroth21) [inline xml/libvirt_configuration_virsh.xml]: ```<literal>eth0</literal>```

File: xml/libvirt_configuration_virsh.xml [hunk 3]
```diff
@@ -0,0 +1,1231 @@
...
+<screen>&lt;network&gt;
+  &lt;name&gt;passthrough&lt;/name&gt;
+    &lt;forward mode='hostdev' managed='yes'&gt;
+      &lt;pf dev='eth0'/&gt;
+    &lt;/forward&gt;
+  &lt;/network&gt;</screen>
+    <para>
+     To use this network on the host, save the above code to a file, for
+     example <filename>/tmp/passthrough.xml</filename>, and execute the
+     following commands. Remember to replace eth0 with the real network
```
Comment (taroth21) [inline xml/libvirt_configuration_virsh.xml]: tag 'eth0' with literal, see above

File: xml/libvirt_configuration_virsh.xml [hunk 4]
```diff
@@ -0,0 +1,1231 @@
...
+     the <xref linkend="vt.io.sriov"/> device from the pool created in
+     <xref linkend="libvirt.config.io.pool.host"/>. <systemitem class="library">libvirt</systemitem> automatically
+     derives the list of all VFs associated with that PF the first time the
+     guest is started.
+    </para>
+<screen>&lt;interface type='network'&gt;
+  &lt;source network='passthrough'&gt;
+&lt;/interface&gt;</screen>
+    <para>
+     To verify the list of associated VFs, run <command>virsh net-dumpxml
```
Comment (taroth21) [inline xml/libvirt_configuration_virsh.xml]: 'To verify the list of associated VFs, run [...] on the host after the first guest that uses the network with the pool of VFs starts.' is hard to read. Maybe rephrase to:
 'After the first guest starts that uses the network with the pool of VFs, verify the list [...]. Do so by running [...] on the host:'

File: xml/libvirt_configuration_virsh.xml [hunk 5]
```diff
@@ -0,0 +1,1231 @@
...
+  &lt;/forward&gt;
+  &lt;/network&gt;</screen>
+   </sect3>
+  </sect2>
+ </sect1>
+
+ <sect1 xml:id="libvirt.config.storage.virsh">
+  <title>Configuring Storage Devices</title>
+  <para>
+  Information about storage devices is defined under the <tag>disk</tag> element.
```
Comment (taroth21) [inline xml/libvirt_configuration_virsh.xml]: 'Information is defined' ... AFAIK, information is a plural word, maybe rephrase to: 'Storage devices are defined within the ```disk``` element.'

File: xml/libvirt_configuration_virsh.xml [hunk 6]
```diff
@@ -0,0 +1,1231 @@
...
+  The usual <tag>disk</tag> element supports several attributes, but the
+   following two attributes are the most important:
```
Comment (taroth21) [inline xml/libvirt_configuration_virsh.xml]: Split in two sentences: 'The usual <tag>disk</tag> element supports several attributes.  The following two attributes are the most important:'

File: xml/libvirt_configuration_virsh.xml [hunk 7]
```diff
@@ -0,0 +1,1231 @@
...
+  </para>
+  <itemizedlist>
+   <listitem>
+    <para>The <tag class="attribute">type</tag> attribute describes the source
+     of the virtual disk device. Valid values are <tag class="attvalue">file</tag>,
+     <tag class="attvalue">block</tag>, <tag class="attvalue">dir</tag>,
+     <tag class="attvalue">network</tag>, or <tag class="attvalue">volume</tag>.</para>
+   </listitem>
+   <listitem>
+    <para>The <tag class="attribute">device</tag> attribute indicates, how
```
Comment (taroth21) [inline xml/libvirt_configuration_virsh.xml]: no comma after 'indicates'

File: xml/libvirt_configuration_virsh.xml [hunk 8]
```diff
@@ -0,0 +1,1231 @@
...
+     <tag class="attvalue">cdrom</tag>, and others.
+    </para>
+   </listitem>
+  </itemizedlist>
+  <para>
+   The following child elements are the most important:
+  </para>
+  <itemizedlist>
+   <listitem>
+    <para><tag>driver</tag> contains the driver and the bus, which is used by
```
Comment (taroth21) [inline xml/libvirt_configuration_virsh.xml]: split in two sentences: '<tag>driver</tag> contains the driver and the bus. Those are used by the VM guest to work with the new disk device.</para>'

File: xml/libvirt_configuration_virsh.xml [hunk 9]
```diff
@@ -0,0 +1,1231 @@
...
+     the VM guest to work with the new disk device.</para>
+   </listitem>
+   <listitem>
+    <para>The <tag>target</tag> element contains the device name, under which
+     the new disk is shown in the VM guest.
+     as well as the optional bus attribute, which defines the type of bus,
```
Comment (taroth21) [inline xml/libvirt_configuration_virsh.xml]: 'as well as' -> 'It also contains the optional'

File: xml/libvirt_configuration_virsh.xml [hunk 10]
```diff
@@ -0,0 +1,1231 @@
...
+        <para>Add a <tag>disk</tag> element inside the <tag>disk</tag> element together with the attributes <tag class="attvalue">type</tag> and
+        <tag class="attvalue">device</tag>.</para>
+        <screen>&lt;disk type='file' device='disk'&gt;</screen>
+      </step>
+      <step>
+        <para>Specify a <tag>driver</tag> element and use the default values:</para>
+        <screen>&lt;driver name='qemu' type='qcow2'/&gt;</screen>
+      </step>
+      <step>
+        <para>Create a disk image, which will be used as a source for the new virtual disk device</para>
```
Comment (taroth21) [inline xml/libvirt_configuration_virsh.xml]: add colon at the end: ':'

File: xml/libvirt_configuration_virsh.xml [hunk 11]
```diff
@@ -0,0 +1,1231 @@
...
+      <step>
+        <para>Restart your VM:</para>
+        <screen><command>sudo virsh start sles15</command></screen>
+      </step>
+  </procedure>
+  <para>Your new storage device should be available in the Guest OS.</para>
+ </sect1>
+
+  <sect1 xml:id="libvirt.video_virsh">
+    <title>Configuring Video devices</title>
```
Comment (taroth21) [inline xml/libvirt_configuration_virsh.xml]: title style: 'Configuring Video Devices'

File: xml/libvirt_configuration_virsh.xml [hunk 12]
```diff
@@ -0,0 +1,1231 @@
...
+    <para>
+    When using the Virtual Machine Manager,
+    only the Video device model can be defined. The amount of allocated VRAM
+    or 2D/3D acceleration can only be changed in the XML configuration.
+    </para>
+    <sect2 xml:id="libvirt.video_vram_virsh">
+      <title>Changing the amount of allocated VRAM</title>
```
Comment (taroth21) [inline xml/libvirt_configuration_virsh.xml]: use title style for headlines (all subsections and tip)

File: xml/libvirt_configuration_virsh.xml [hunk 13]
```diff
@@ -0,0 +1,1231 @@
...
+
+    <sect2 xml:id="libvirt.video_accel_virsh">
+      <title>Changing the state of 2D/3D acceleration</title>
+      <procedure>
+        <step>
+        <para>Edit the configuration for an existing VM Guest:</para>
+        <screen><prompt>tux &gt; </prompt><command>sudo virsh edit sles15</command></screen>
+      </step>
+      <step>
+        <para>To enable/disable 2D/3D acceleration, change the value of 'accel3d' and 'accel2d'
```
Comment (taroth21) [inline xml/libvirt_configuration_virsh.xml]: 'accel3d' and 'accel2d' -> tag with ```<literal>```

File: xml/libvirt_configuration_virsh.xml [hunk 14]
```diff
@@ -0,0 +1,1231 @@
...
+...
+&lt;/model&gt;
+&lt;/video&gt;
+        </screen>
+      </step>
+      </procedure>
+      <tip>
+        <title>Note on enabling 2D/3D acceleration</title>
+        <para>
+        Only 'vbox' video devices are capable of 2D/3D acceleration.
```
Comment (taroth21) [inline xml/libvirt_configuration_virsh.xml]: 'vobx' -> tag with ```<literal/>```

**Comment by bbevan** [comment]
Should someone make the requested changes so this can be merged? 

@taroth21

File: xml/libvirt_configuration_virsh.xml [hunk 15]
```diff
@@ -0,0 +1,1231 @@
...
+]>
+<chapter xmlns="http://docbook.org/ns/docbook" xmlns:xi="http://www.w3.org/2001/XInclude" xmlns:xlink="http://www.w3.org/1999/xlink" xml:base="libvirt_configuration_virsh.xml" version="5.0" xml:id="cha.libvirt.config_virsh">
+ <title>Configuring Virtual Machines with virsh</title>
+ <info>
+  <abstract>
+   <para>
+    In addition to the Virtual Machine Manager, the VM guest can also be configured with <command>virsh</command>.
+    <command>virsh</command> is a command line tool to manage virtual machines based on virtualization infrastructures such as
+    KVM, Xen, VMware, or LXC.
+    <command>virsh</command> can control the state of a VM, edit a the configuration of a VM or even migrate a VM to another host.
```
[the remark this answers is not in the document]
Comment (bbevan) [reply xml/libvirt_configuration_virsh.xml]: Proposed rewrite:

"You can use <command>virsh</command> in addition to using the Virtual Machine Manager's ability to configure the VM guest. <command>virsh</command> is a command line tool that manages virtual machines created by VM software, such as KVM, Xen, VMWare, and LXC. <command>virsh</command> can also control the state of a VM, edit its configuration, or even migrate it to another host."

[the remark this answers is not in the document]
Comment (bbevan) [reply xml/libvirt_configuration_virsh.xml]: Rewrite 2:

"You can use <command>virsh</command> to configure the VM guest in addition to using the Virtual Machine Manager. virsh is a command line tool that manages virtual machines created by VM software, such as KVM, Xen, VMWare, and LXC. virsh can also control the state of a VM, edit its configuration, or even migrate it to another host."

That's a little better.

File: xml/libvirt_configuration_virsh.xml [hunk 16]
```diff
@@ -0,0 +1,1231 @@
...
+    The following section shows you the configuration of a VM with <command>virsh</command>, for example CPU, Memory, Input devices or Video devices.
```
[the remark this answers is not in the document]
Comment (bbevan) [reply xml/libvirt_configuration_virsh.xml]: "The following section shows the configuration of a VM with <command>virsh</command>. Examples include modifying memory allocation, input devices, cpu settings, and video configuration."

**Comment by sknorr** [comment]
@bbevan Feel free to add fixes - @lightlace says he doesn't mind.

File: xml/libvirt_configuration_virsh.xml (hunk 15 above)
[the remark this answers is not in the document]
Comment (taroth21) [reply xml/libvirt_configuration_virsh.xml]: @bbevan : Thanks! I used your second proposal. :) But I still changed the last sentence to 'With virsh [...]' because from a linguistic POV, virsh is inanimate and cannot be the agent of a sentence.

File: xml/libvirt_configuration_virsh.xml (hunk 16 above)
[the remark this answers is not in the document]
Comment (taroth21) [reply xml/libvirt_configuration_virsh.xml]: @bbevan : Thanks, used your proposal.

File: xml/libvirt_configuration_gui.xml [hunk 17]
```diff
@@ -534,78 +534,219 @@
...
  <sect1 xml:id="sec.libvirt.config.input">
   <title>Input Devices</title>
+	<para>
+		This section focuses on adding and configuring new input devices such as Mouse, Keyboard or Tablet.
+	</para>
+  <procedure>
+   <title>Adding a New Input Device</title>
+   <step>
+    <para>
+     Click <guimenu>Add Hardware</guimenu> below the left panel, then select
```
[the remark this answers is not in the document]
Comment (taroth21) [reply xml/libvirt_configuration_gui.xml]: Fixed throughout the book with 5d374e2204d6dc91.

**Comment by cwickert** [comment]
I took the liberty of fixing the conflicts so we can finally merge, but when comparing the old sec.libvirt.config.mahcinetype.virsh (note the typo) with your new sec.libvirt.config.machinetype.virsh, I realized that there were some problems such as the prompts I fixes in 44b19b5e. And there are more problems that should be fixed first, please wait for my conclusive review tomorrow.

**Comment by cwickert** [comment]
Ah, I merged against an outdated 'develop' branch and there is still a conflict. :-( Let me fix this first.

**Comment by fsundermeyer** [comment]
@cwickert : Please fix the issues rather than commenting since lightlace has moved to another project.
